CONTRIBUTING.md

```diff
@@ -141,51 +144,7 @@ We also use Github Actions CI to run tests on every branch.
 
 The release process is semi-automated via github actions. A number of steps are necessarily left manual (such as versioning) and require intervention from the user.
 
-Follow the steps outline in [Package Release Process for Measures Data](https://confluence.cms.gov/x/jmNiP) to publish new version.
```

Comment:
To my knowledge, these logging and debugging docs have not proved useful for us. Removed to clean up clutter.